bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/graphics/FontMetrics.java

```diff
@@ -31,6 +31,21 @@ public final class FontMetrics {
 FontMetrics() {
 }
 
+/**
+ * Convenience method to make a copy of receiver.
+ *
+ * @since 3.125
+ */
+public FontMetrics clone () {
```

Comment:
Is this a new API in FontMetrics? In that case shouldn't it be added to all platforms?

Reply:
I had no intention to add a cross-platform API.
Just wanted a convenience method to avoid copy&pasting code.

Reply:
Does public part have to match on all platforms?
If yes, would you rather have the method on other platforms, or removed from macOS via moving it into `TextLayout` as a private helper?

Reply:
> Does public part have to match on all platforms?

Yes, otherwise there is a strong risk that the code written on a particular platform starts using those methods, and that as a result such code becomes non-portable, which is the opposite reason of why one would use SWT.

> If yes, would you rather have the method on other platforms, or removed from macOS via moving it into TextLayout as a private helper?

Can't the method be only package visible here?

Reply:
Ah right, I was confused because `StyledText` is in a different package. Will make it package-private then.

Reply:
> Yes, otherwise there is a strong risk that the code written on a particular platform starts using those methods, and that as a result such code becomes non-portable, which is the opposite reason of why one would use SWT.

+1

Reply:
> Yes, otherwise there is a strong risk that the code written on a particular platform starts using those methods, and that as a result such code becomes non-portable, which is the opposite reason of why one would use SWT.

To be fair we have this situation already with Windows OLE API .... in the end it might be better to throw unsupported operation and have the API everywhere as this is really a nightmare to develop.

Reply:
> To be fair we have this situation already with Windows OLE API ....

Win OLE api is in a platform specific package `org.eclipse.swt.ole.win32`, so all the classes in the package are platform specific -  https://help.eclipse.org/latest/index.jsp?topic=/org.eclipse.platform.doc.isv/reference/api/org/eclipse/swt/ole/win32/package-summary.html

Reply:
done